egograph/backend/infrastructure/database/query_params.py

```
@@ -0,0 +1,85 @@
+
+def execute_query(
+    conn: duckdb.DuckDBPyConnection, sql: str, params: list[Any] | None = None
+) -> list[dict[str, Any]]:
+    """SQLクエリを実行し、結果を辞書のリストとして返します。
+
+    numpy/pandas型をPython標準型に変換することで、JSONシリアライズに対応します。
+
+    Args:
+        conn: DuckDBコネクション
+        sql: 実行するSQLクエリ
+        params: SQLパラメータ（オプション）
```

Ruff (0.15.13)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)